pkg/cfn/builder/vpc_endpoint.go

```diff
@@ -123,16 +123,27 @@ func (e *VPCEndpointResourceSet) hasEndpoint(endpoint string) bool {
 	return false
 }
 
-// BuildVPCEndpointServices builds a slice of VPCEndpointServiceDetails for the specified endpoint names
-func BuildVPCEndpointServices(ec2API ec2iface.EC2API, region string, endpoints []string) ([]VPCEndpointServiceDetails, error) {
+var chinaPartitionServiceHasChinaPrefix = map[string]bool{
+	api.EndpointServiceEC2:            true,
+	api.EndpointServiceECRAPI:         true,
+	api.EndpointServiceECRDKR:         true,
+	api.EndpointServiceS3:             false,
+	api.EndpointServiceSTS:            true,
+	api.EndpointServiceCloudFormation: true,
+	api.EndpointServiceAutoscaling:    true,
+	api.EndpointServiceCloudWatch:     false,
+}
+
+// buildVPCEndpointServices builds a slice of VPCEndpointServiceDetails for the specified endpoint names
+func buildVPCEndpointServices(ec2API ec2iface.EC2API, region string, endpoints []string) ([]VPCEndpointServiceDetails, error) {
 	serviceNames := make([]string, len(endpoints))
-	servicePrefix := ""
-	if api.Partition(region) == api.PartitionChina {
-		servicePrefix = "cn."
-	}
-	serviceRegionPrefix := fmt.Sprintf("%scom.amazonaws.%s.", servicePrefix, region)
+	serviceDomain := fmt.Sprintf("com.amazonaws.%s", region)
 	for i, endpoint := range endpoints {
-		serviceNames[i] = serviceRegionPrefix + endpoint
+		name := fmt.Sprintf("%s.%s", serviceDomain, endpoint)
+		if api.Partition(region) == api.PartitionChina && chinaPartitionServiceHasChinaPrefix[endpoint] {
```

Comment:
Based on the above comment, this would become:
```suggestion
		if api.Partition(region) == api.PartitionChina && !chinaPartitionServicesWithoutPrefix[endpoint] {
```